lib/speech/index.js

```diff
@@ -0,0 +1,903 @@
+    callback(null, self.operation(apiResponse, decoder), apiResponse);
+  });
+};
+
+/**
+ * Asynchronously detect speech from an audio file.
+ *
+ * @param {string|module:storage/file} file - The source file to run the
+ *     detection on. It can be either a local file path, a remote file URL, a
+ *     Cloud Storage URI, or a gcloud File object.
+ * @param {object} recognitionConfig - A
+ *     `RecognitionConfig` object. See [`RecognitionConfig`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig).
+ * @param {string} recognitionConfig.encoding - Encoding of audio data
+ *     sent in all `RecognitionAudio` messages. See
+ *     [`RecognitionConfig.encoding`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ * @param {number} recognitionConfig.sampleRate - Sample rate in Hertz
+ *     of the audio data sent in all RecognitionAudio messages. See
+ *     [`RecognitionConfig.sampleRate`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ * @param {string=} recognitionConfig.languageCode - See
+ *     [`RecognitionConfig.languageCode`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `"en-US"`.
+ * @param {number=} recognitionConfig.maxAlternatives - See
+ *     [`RecognitionConfig.maxAlternatives`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `1`.
+ * @param {boolean=} recognitionConfig.profanityFilter - See
+ *     [`RecognitionConfig.profanityFilter`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `false`.
+ * @param {string[]=} recognitionConfig.speechContext - See
+ *     [`RecognitionConfig.speechContext`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `[]`.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:speech/operation} callback.operation - `Operation` object. See
+ *     [`Operation`](https://cloud.google.com/speech/reference/rpc/google.longrunning#google.longrunning.Operation).
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * //-
+ * // <h3>Run speech detection over a local file</h3>
+ * //-
+ * speech.createDetectJob('./bridge.raw', {
+ *   encoding: 'LINEAR16',
+ *   sampleRate: 16000
+ * }, function(err, operation, apiResponse) {
+ *   setTimeout(function() {
+ *     operation.get(function(err, operation) {
+ *       // operation.response.results[0].alternatives[0] = {
```

Comment:
I think a longhand example here would be better, e.g.

```js
// {
//   response: {
//     results: [
//       {
//         altneratives: [
//           {
//             transcript: '...',
//             confidence: 0.88
//           }
//         ]
//       }
//     ]
//   }
// }
```

Also, why `setTimeout`?